evm-contracts/test/v0.7/Operator.test.ts

```diff
@@ -705,6 +712,908 @@ describe('Operator', () => {
+
+        it('throws an error', async () => {
+          await matchers.evmRevert(async () => {
+            await link.transferAndCall(operator.address, paid, maliciousData)
+          })
+        })
+      })
+    })
+
+    describe('when not called through the LINK token', () => {
+      it('reverts', async () => {
+        await matchers.evmRevert(async () => {
+          await operator
+            .connect(roles.oracleNode)
+            .oracleRequest2(
+              '0x0000000000000000000000000000000000000000',
+              0,
+              specId,
+              to,
+              fHash,
+              1,
+              1,
+              '0x',
+            )
+        })
+      })
+    })
+  })
+
+  describe('#fulfillOracleRequest2', () => {
+    describe('single word fulfils', () => {
+      const response = 'Hi mom!'
+      let maliciousRequester: contract.Instance<MaliciousRequesterFactory>
+      let basicConsumer: contract.Instance<BasicMultiWordConsumerFactory>
+      let maliciousConsumer: contract.Instance<MaliciousMultiWordConsumerFactory>
+      let gasGuzzlingConsumer: contract.Instance<GasGuzzlingMultiWordConsumerFactory>
+      let request: ReturnType<typeof oracle.decodeRunRequest>
+
+      describe('gas guzzling consumer', () => {
+        beforeEach(async () => {
+          gasGuzzlingConsumer = await gasGuzzlingConsumerFactory
+            .connect(roles.consumer)
+            .deploy(link.address, operator.address, specId)
+          const paymentAmount = h.toWei('1')
+          await link.transfer(gasGuzzlingConsumer.address, paymentAmount)
+          const tx = await gasGuzzlingConsumer.gassyRequestEthereumPrice(
+            paymentAmount,
+          )
+          const receipt = await tx.wait()
+          request = oracle.decodeRunRequest(receipt.logs?.[3])
+        })
+
+        it('emits an OracleResponse2 event', async () => {
+          const fulfillParams = oracle.convertFufillParams(request, response)
+          const tx = await operator
+            .connect(roles.oracleNode)
+            .fulfillOracleRequest2(...fulfillParams)
+          const receipt = await tx.wait()
+          assert.equal(receipt.events?.length, 1)
+          const responseEvent = receipt.events?.[0]
+          assert.equal(responseEvent?.event, 'OracleResponse2')
+          assert.equal(responseEvent?.args?.[0], request.requestId)
+          assert.equal(responseEvent?.args?.[5], fulfillParams[5])
+        })
+      })
+
+      describe('cooperative consumer', () => {
+        beforeEach(async () => {
+          basicConsumer = await basicMultiWordConsumerFactory
+            .connect(roles.defaultAccount)
+            .deploy(link.address, operator.address, specId)
+          const paymentAmount = h.toWei('1')
+          await link.transfer(basicConsumer.address, paymentAmount)
+          const currency = 'USD'
+          const tx = await basicConsumer.requestEthereumPrice(
```

Comment:
I think this can be a little misleading, while the text "Hi Mom!" itself could fit into a single EVM slot, it does not because the callback function signature for this fulfillment is `fulfill(bytes32 _requestId, bytes memory _price)`. A `bytes` type [always takes up at least 2 words](https://solidity.readthedocs.io/en/v0.7.0/abi-spec.html#use-of-dynamic-types), the first word denote the length length of the payload and then at least one additional word containing the actual payload(sometimes it's all zeros if the length is zero). What we actually want is to be able to respond to `fulfill(bytes32 _requestId, bytes32 _price)` in the new fulfillment workflow. I'll leave some notes there because this may not be straightforward.

Reply:
Great spot. I've altered the test to use a `bytes32` consumer and it fails to fulfil the data.